workbench-core/accounts/src/utilities/hostingAccountLifecycleService.ts

```diff
@@ -479,6 +505,17 @@ export default class HostingAccountLifecycleService {
     return currOutputVal;
   }
 
+  private _getEnvironmentFilesPathForArn(artifactBucketArn: string): string {
+    const parsedBucketArn = artifactBucketArn.replace('arn:aws:s3::::', '').split('/');
+    const bucketName = parsedBucketArn[0];
+
+    if (_.isEmpty(bucketName) || bucketName.length === 0) {
+      Boom.badRequest(`Could not identify bucket name in S3 ARN ${artifactBucketArn}`);
```

Comment:
I think you have to throw the boom error for it to propogate. Also I think this should be a `Boom.internalServerError`, since the `artifactBucketArn` is not coming from the user request, but from the CFN output. Some how the main account was set up incorrectly and outputed the wrong bucket arn for the bucket.

Reply:
Will do that and add a test for coverage.

Reply:
Added!